src/gerbil/gxi

```diff
@@ -1,11 +1,48 @@
 #!/bin/sh
 set -eu
 
+
 if [ -z "${GERBIL_HOME:-}" ]; then
-    GERBIL_HOME=$(dirname $(cd ${0%/*} && echo $PWD))
+
+    # First, try whereis. We are looking for $GERBIL_HOME/lib/std, our stdlib
+    for gh in $(whereis -b gerbil |cut -d: -f2); do
+        if [ -d $gh/lib/std ]; then
+            GERBIL_HOME="$gh" ; break ;
+        fi;
+    done
+
+    # if that did not work, find out where we are and try that.
+    if [ -z "${GERBIL_HOME:-}" ]; then
+        GERBIL_BIN_LOCATION="$0"
+
+        # Check if we're a symlink
+        GERBIL_LINK=$(readlink $GERBIL_BIN_LOCATION || true)
+        while true
+        do
+            # If not, "../" is our home
+            if [ -z "$GERBIL_LINK" ]; then
+                GERBIL_HOME=$(dirname $(cd ${GERBIL_BIN_LOCATION%/*} && echo $PWD))
```

Comment:
Definitions don't need "" on the right side, but all other uses need it, including this ${...} and $PWD. Moreover, printf is usually better than echo in case variables start with `-`, which hopefully isn't the case for `$PWD`.